src/nat/front_ends/fastapi/api_versioning.py

```
@@ -0,0 +1,71 @@
+    def __init__(self, *, api_version_header: bool = True, version: str = "1"):
+        self.api_version_header = api_version_header
+        self.version = version
+
+    @classmethod
+    def from_config(cls, cfg: FastApiFrontEndConfig):  # pragma: no cover - simple wiring helper
+        version_cfg = getattr(cfg, "versioning", None)
+        return cls(api_version_header=bool(getattr(version_cfg, "api_version_header", True)),
+                   version=str(getattr(version_cfg, "version", 1)))
+
+
+def apply_version_headers(response: Response, opts: VersioningOptions, *, is_legacy: bool = False) -> None:
```

Ruff (0.14.13)

Unused function argument: `is_legacy`

(ARG001)